### PR TITLE
PE-930: validate file name when uploading files

### PR DESCRIPTION
**Pull request by karlprieb**
This PR is heavily based on @matibat work on rename files, folders and drives feature and #143 

- improves ArFS entity name validators tests making them parameterized
- add drive file validation to `uploadPublicFIle` and `uploadPrivateFile` integration tests
- add file name validation to file upload method

### CODE REVIEW

File: src/ardrive.ts [hunk 1]
```diff
@@ -440,6 +441,8 @@ export class ArDrive extends ArDriveAnonymous {
 			communityTipSettings: CommunityTipSettings
 		) => Promise<ArFSUploadPublicFileResult | ArFSUploadPrivateFileResult>
 	): Promise<ArFSResult> {
+		assertValidArFSFileName(wrappedFile.getBaseFileName());
```
Comment (fedellen) [inline src/ardrive.ts:444]: This should be moved below this line and use `destinationBaseName`:

```ts
wrappedFile.newFileName = destinationFileName;

// Assert based on intended destination name
assertValidArFSFileName(wrappedFile.destinationBaseName)

const nameConflictInfo = await getConflictInfoFn(parentFolderId);

```

`getBaseFileName()` will only return the base file name from disk as opposed to the user's intended file name. Sorry for the confusing api there. That one should probably be a private method and the names could be much clearer.

Comment (karlprieb) [reply src/ardrive.ts:444]: It make sense. Thank you! :+1:

**Review by matibat — COMMENTED** [review]
Looking good

**Note**
In the future, we might get rid of some of the reserved characters ending only with the Unix ones (e.i. NULL and "/") which are not allowed in the file name. At that moment we might need to stub the `fs` lib to let it return names with reserved characters or maybe dismiss that test.
_This note is not a change request_

File: src/ardrive.ts (hunk 1 above)
Comment (karlprieb) [reply src/ardrive.ts]: Fixed d88bb917bb2e2e2c15a0ac3a4b5b6ccea710b5aa

**Review by fedellen — APPROVED** [review]
👍

File: src/arfs/arfs_entity_name_validators.test.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import { expect } from 'chai';
+
```
Comment (arielmelendez) [inline src/arfs/arfs_entity_name_validators.test.ts:2]: Whitespace PR spam

**Review by arielmelendez — APPROVED** [review]

**Pull request merged**